snap/snapcraft.yaml

```diff
@@ -121,48 +109,46 @@ architectures:
   - build-on: ppc64el
 
 parts:
-
   uros-agent:
     plugin: colcon
     source: .
-    override-build: |
-        set +u
-        git clone https://github.com/eProsima/Micro-XRCE-DDS-Agent.git -b $ROS_DISTRO
-        git clone https://github.com/eProsima/Micro-CDR.git -b $ROS_DISTRO
-        git clone https://github.com/micro-ROS/micro_ros_msgs.git -b $ROS_DISTRO
-        git clone https://github.com/micro-ROS/rosidl_typesupport_microxrcedds.git -b $ROS_DISTRO
-        git clone https://github.com/micro-ROS/rmw-microxrcedds.git -b $ROS_DISTRO
-        git clone https://github.com/micro-ROS/micro-ROS-Agent.git -b $ROS_DISTRO
-        . /opt/ros/$ROS_DISTRO/setup.sh
-        colcon build --merge-install --install-base $SNAPCRAFT_PRIME --cmake-args "-DUAGENT_BUILD_EXECUTABLE=OFF -DUAGENT_USE_SYSTEM_FASTDDS=ON" --packages-up-to micro_ros_agent
-        set -u
+    override-pull: |
+        snapcraftctl pull
+        [ ! -d Micro-XRCE-DDS-Agent ] && git clone https://github.com/eProsima/Micro-XRCE-DDS-Agent.git -b $ROS_DISTRO
+        # Place additional colcon.pkg for correct topological order
+        mv snap/local/micro_ros_agent_colcon.pkg micro_ros_agent/colcon.pkg
+
+        version="$(git describe --always --tags| sed -e 's/^v//;s/-/+git/;y/-/./')"
+        [ -n "$(echo $version | grep "+git")" ] && grade=devel || grade=stable
+        snapcraftctl set-version "$version"
+        snapcraftctl set-grade "$grade"
+
+    colcon-cmake-args:
+      - -DUAGENT_BUILD_EXECUTABLE=OFF
+      - -DUAGENT_USE_SYSTEM_FASTDDS=ON
+      - -DMICROROSAGENT_SUPERBUILD=OFF
```

Comment:
The reason I turned it off was because when rebuilding CMake fails to find microxrcedds_agent
```
CMake Error at CMakeLists.txt:34 (find_package):
  By not providing "Findmicroxrcedds_agent.cmake" in CMAKE_MODULE_PATH this
  project has asked CMake to find a package configuration file provided by
  "microxrcedds_agent", but CMake did not find one.
```
Forcing the clean and build from scratch. Would you have a way to support re-build with `-DMICROROSAGENT_SUPERBUILD=ON` ?

Reply:
Would you still prefer the SUPERBUILD even if we cannot rebuild?

Reply:
I was thinking about building on a clean environment each time. 
Can we achieve this? Maybe delete the colcon build files?

Sorry for the delay BTW.

Reply:
Forcing the execution of the `ExternalProject_Add` of xrceagent from the SuperBuild is working for the rebuild.
Since the first build is turning off the MICROROSAGENT_SUPERBUILD option, I just have to rewrite it (specifying `--cmake-force-configure`).
It makes the snapcraft.yaml cleaner, and I could remove the colcon.pkg.